graphql-dgs-codegen-shared-core/src/main/kotlin/com/netflix/graphql/dgs/client/codegen/GraphQLMultiQueryRequest.kt

```diff
@@ -0,0 +1,90 @@
+/*
+ *
+ *  Copyright 2020 Netflix, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+package com.netflix.graphql.dgs.client.codegen
+
+import graphql.language.*
+
+class GraphQLMultiQueryRequest(
+    private val requests: List<GraphQLQueryRequest>
+) {
+
+    fun serialize(): String {
+        if (requests.isEmpty()) throw AssertionError("Request must have at least one query")
+
+        if (requests.size == 1) {
+            return requests[0].serialize()
+        }
+
+        return multiQuery()
+    }
+
+    private fun multiQuery(): String {
+        val operationDef = OperationDefinition.newOperationDefinition()
+        requests[0].query.name?.let { operationDef.name(it) }
```

Comment:
Add a check to ensure there is at least one request.

Reply:
Function multiQuery() is private, meaning you have to call serialize to hit it. In the serialize() function on line 28, I check for at least one request. Is that sufficient, or is another check needed?

Reply:
I see missed that earlier, we can skip it in that case. Although, I can see someone flipping that  to  be `public` and forgetting to add that check.

Reply:
I just combined the multiQuery() and serialize() methods together to avoid someone causing problems by making multiQuery() public.